### PR TITLE
Improve join phases labels and help messages

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR fixes command labels/help texts for kubeadm join phases

**Which issue(s) this PR fixes**:
none

**Special notes for your reviewer**:
This PR assumes discovery flags will apply as proposed in https://github.com/kubernetes/kubernetes/pull/74831

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig cluster-lifecycle
/area kubeadm
/priority important-soon
@kubernetes/sig-cluster-lifecycle-pr-reviews
/assign @neolit123 
/cc @ereslibre 
/cc @rosti 
/cc @yagonobre

### CODE REVIEW

File: cmd/kubeadm/app/cmd/phases/workflow/runner.go [hunk 1]
```diff
@@ -442,8 +442,11 @@ func (e *Runner) prepareForExecution() {
 // addPhaseRunner adds the phaseRunner for a given phase to the phaseRunners list
 func addPhaseRunner(e *Runner, parentRunner *phaseRunner, phase Phase) {
 	// computes contextual information derived by the workflow managed by the Runner.
-	generatedName := strings.ToLower(phase.Name)
-	use := generatedName
+	use := strings.ToLower(phase.Name)
+	if pos := strings.Index(use, " "); pos != -1 {
+		use = use[:pos]
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/workflow/runner.go]: if we don't include the `[api-server-endpoint]` as part of the phase names we can avoid the hack. :(
would it suffice to have the `[api-server-endpoint]` as part of `Example:`?

**Review by neolit123 — COMMENTED** [review]
thanks @fabriziopandini 
i think the edits re OK.
added one suggestion about `[api-server-endpoint]`.

**Comment by neolit123** [comment]
where can i help with the join phases?
i can re-do this PR if you want or take something else?

File: cmd/kubeadm/app/cmd/phases/workflow/runner.go (hunk 1 above)
Comment (ereslibre) [reply cmd/kubeadm/app/cmd/phases/workflow/runner.go]: Maybe we could add a specific field to `Phase` struct to make this logic cleaner?

Comment (fabriziopandini) [reply cmd/kubeadm/app/cmd/phases/workflow/runner.go]: I don' think we should overengineer this.
having args descriptor as part of the name is usual in cobra (nb phases `Name` maps to cobra `Use` attribute) , and it makes a nice "Usage" (prior art e.g. kubectl). in this case
```
Usage:
  kubeadm join [api-server-endpoint] [flags]
```
The logic above, that now I have refactored in a separated `cleanName` function and documented, starts from what we are passing to cobra and returns a cleaned version that we are going to use in the list of phases

**Comment by fabriziopandini** [comment]
@neolit123 help is always welcome! 
IMO this PR is ok. I'm updating the tracking issue https://github.com/kubernetes/kubeadm/issues/1204 with what is my list/what is resulted from "Auditing Join Phases" document

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: cmd/kubeadm/app/cmd/phases/workflow/runner.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/workflow/runner.go]: ok

File: cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go [hunk 2]
```diff
@@ -74,8 +74,8 @@ func getControlPlanePreparePhaseFlags() []string {
 
 func newControlPlanePrepareDownloadCertsSubphase() workflow.Phase {
 	return workflow.Phase{
-		Name:  "download-certs",
-		Short: fmt.Sprintf("Download certificates from %s", kubeadmconstants.KubeadmCertsSecret),
+		Name:  "download-certs [api-server-endpoint]",
+		Short: fmt.Sprintf("[EXPERIMENTAL] Downloads certificates shared among control-plane nodes from the %s Secret", kubeadmconstants.KubeadmCertsSecret),
 		Long:  cmdutil.MacroCommandLongDescription,
```
Comment (rosti) [inline cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go]: `MacroCommandLongDescription` should be removed. It generates wrong and misleading help message.

**Review by rosti — COMMENTED** [review]
Thanks @fabriziopandini !

File: cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go (hunk 2 above)
Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go]: i will do the second pass on join phases and clean this up.

Comment (fabriziopandini) [reply cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go]: Thanks. I missed this one

Comment (fabriziopandini) [reply cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go]: Fixed while rebasing

Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go]: /lgtm

**Pull request merged**